onadata/apps/messaging/signals.py

```diff
@@ -23,11 +23,13 @@ def messaging_backends_handler(sender, **kwargs):  # pylint: disable=W0613
     created = kwargs.get('created')
     instance = kwargs.get('instance')
     if instance and created:
-        instance.save()
         for name in backends:
             backend = backends[name]['BACKEND']
             backend_options = backends[name].get('OPTIONS')
             if as_task:
-                call_backend_async.delay(backend, instance.id, backend_options)
+                # Sometimes the Action isn't created yet, hence
+                # the need to delay 2 seconds
+                call_backend_async.apply_async(
+                    (backend, instance.id, backend_options), countdown=2)
```

Comment:
Why are the actions not created at this point sometimes? _Expecting this function to only be called after the Action is saved..._

Reply:
I don't know. I expected the `@use_master` [here](https://github.com/onaio/onadata/blob/master/onadata/apps/messaging/backends/base.py#L13) decorator had this covered. [This](https://github.com/onaio/onadata/blob/master/onadata/apps/messaging/backends/base.py#L19) returned None when called on submission creation and submission edit but had an instance on when called in other places. Hence a message couldn't be sent [here](https://github.com/onaio/onadata/blob/master/onadata/apps/messaging/backends/base.py#L24) on submission creation or edit.

Reply:
Mhh weird... I'm guessing the `created` boolean was also `True` at this point?